packages/docusaurus-plugin-content-blog/src/blogUtils.ts

```diff
@@ -175,14 +174,11 @@ export async function generateBlogPosts(
 
       // Use file create time for blog.
       date = date || (await fs.stat(source)).birthtime;
-      const formattedDate = getDateTimeFormat(i18n.currentLocale)(
-        i18n.currentLocale,
-        {
-          day: 'numeric',
-          month: 'long',
-          year: 'numeric',
-        },
-      ).format(date);
+      const formattedDate = new Intl.DateTimeFormat(i18n.currentLocale, {
```

Comment:
We didn't use Intl directly here because we didn't want to drop Node 12 support yet.
Afaik full icu is only supported since Node 13. 

We need Node 12 + the ability to format dates in any locale in node, do you think it will still work without the polyfill?

cc @lex111

Reply:
Yes, that's right, probably it was necessary to write a comment in the code to remove possible misunderstandings.

Reply:
So full-icu is the only option here. intl.js has a very old version of cldr and is not spec compliant as of a few years ago. Couple of questions:

What portion of docusaurus runs in node?
What locales are actually supported? And don't say all because that changes per cldr version 😃

Reply:
> intl.js has a very old version of cldr and is not spec compliant as of a few years ago. 

Can this be "good enough", considering most users will only use English or Node 14, and maybe stale CLDR data will work most of the time?  If this does not we can just ask them to use Node 14 instead.

> What portion of Docusaurus runs in node?

All the content plugins run in node, compute the props to pass to components, and then the theme components + theme-common code are used in SSR + client.

> What locales are actually supported? And don't say all because that changes per cldr version 😃

We don't provide any list of supported locale, if the user pass `{i18n: {defaultLocale: "fr"}}` in the site config, we'll try to pass that locale/language to the Intl APIs, so basically any locale can be used.

> So full-icu is the only option here. 

If the user is really willing to use Node 12 + a locale that is not properly supported in the polyfill, can't we just tell him to use Node 14, or install full ICU on Node 12? (as it will be used in priority over the polyfill)

Reply:
We could also require users to upgrade to Node 14 if they are using any i18n locale != "en" (default)? 

I  think this is reasonable to ask Docusaurus i18n uses to be on Node 14, while other regular users can keep using Node 12?

Reply:
Yeah i think that's fair. Node 13+ for i18n otherwise Node 12 is fine. Might need to adjust CI though to run i18n tests on node 14

Reply:
I'm happy to make the necessary changes if that's the consensus.

Reply:
I think that would be ok to bump .nvmrc to node 14 and a few CI scripts too.

But we should probably keep at least once CI on Node 12 at least until we plan to drop it for all users.

It seems this PR still builds on Node 12 so maybe it's not a big deal and we don't need to upgrade the CI?

Reply:
Sgtm

Reply:
@slorber wanna take another look?